### DISCUSSION TITLE
The true position of an object in the camera in bathing task

### DISCUSSION THREAD

**Discussion by TomatoPicker278** [empriselab/RCareWorld#166] [Other Q&A] [answered]
Sorry, I have solved the problem, but I don't know how to delete this post.

**Comment by YoruCathy** [comment]
Hi @TomatoPicker278 would you mind posting your original question and the solution here in case other people have similar questions?

**Reply by TomatoPicker278** [reply, marked as answer]
OK, no problem. I'll briefly restate the question.

Question: For the camera in the `CameraAttr` attribute, is there an existing API to obtain the real position of an object in the camera coordinate system? (Note: While the camera in the `ActiveLightSensorAttr` attribute has the `GetActiveDepth()` function for retrieving the real depth, it still cannot provide the real 3D coordinates in the camera coordinate system, and it requires creating a new camera. This question is specifically about how to obtain the real position of an object in the camera coordinate system using only the camera in `CameraAttr`.)

My solution: There is no ready-made API available, so the only option is to use the most basic method. Specifically, this involves using `GetDepth()` to get the depth values, then converting these numerical values to real depth values based on the defined depth range, and finally using the Intrinsic Matrix to calculate the real 3D position of a specific RGB pixel. Notably, the Intrinsic Matrix also needs to be specified separately.

This completes the task of obtaining the real position of an object in the camera coordinate system.
